src/main/java/seedu/parser/Parser.java

```diff
@@ -169,12 +173,32 @@ private static Command prepareScore(String[] arguments) throws EscException {
         return new ScoreCommand(subjectIndex);
     }
 
+    /**
+     * Returns the number of questions to quiz from the user input.
+     * @return the number of questions to quiz.
+     * @throws EscException if the input number is a non-integer.
+     */
+    private static int getNumberToQuiz(String argument) throws EscException {
+        String num = argument.split(" ")[1];
+        int numToQuiz = 0;
+        try {
+            numToQuiz = Integer.parseInt(num);
+        } catch (NumberFormatException e) {
+            throw new EscException("Number of questions to quiz has to be an integer.");
+        }
+        if (numToQuiz == 0) {
+            throw new EscException("Number of questions to quiz has to be a positive integer.");
+        }
+        return numToQuiz;
+    }
+
     /**
      * Returns the subject index from the user input.
      * @return the subject index integer.
      * @throws EscException if the subject index is absent or non-integer.
      */
     private static int getSubjectIndex(String argument) throws EscException {
+        argument = argument.split(" ")[0];
```

Comment:
@alexlim510 please help review this file/change! I'm not sure if this is done correctly or not

Reply:
@JeremiasLiew I don't quite get what's happening here aha what's this for?